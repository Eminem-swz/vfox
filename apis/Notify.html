<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Notify 通知类型弹窗 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.82adce24.css">
    <link rel="modulepreload" href="/vfox/assets/Home.d95ceb94.js">
    <link rel="modulepreload" href="/vfox/assets/app.a39fe1d2.js">
    <link rel="modulepreload" href="/vfox/assets/apis_Notify.md.36092315.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.a39fe1d2.js">
    <meta name="twitter:title" content="Notify 通知类型弹窗 | Vfox for Vue3">
    <meta property="og:title" content="Notify 通知类型弹窗 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/LocalStorage" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/LocalStorage" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/LocalStorage">LocalStorage</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/apis/Notify">Notify</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#notify-shownotify-object">Notify.showNotify(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#params">Params</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage">Usage</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#notify-hidenotify-object">Notify.hideNotify([object])</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#params-1">Params</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage-1">Usage</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Scroll">Scroll</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Preview">Preview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Popover">Popover</a><!----></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="display:none;width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="notify-通知类型弹窗" tabindex="-1">Notify 通知类型弹窗 <a class="header-anchor" href="#notify-通知类型弹窗" aria-hidden="true">#</a></h1><p>注：目前仅支持单例模式。</p><h2 id="notify-shownotify-object" tabindex="-1">Notify.showNotify(object) <a class="header-anchor" href="#notify-shownotify-object" aria-hidden="true">#</a></h2><p>显示消息提示。</p><h3 id="params" tabindex="-1">Params <a class="header-anchor" href="#params" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>是</td><td>提示内容</td></tr><tr><td>type</td><td>string</td><td>&#39;default&#39;</td><td>否</td><td>提示类型，可选值：&#39;primary&#39; / &#39;success&#39; / &#39;warning&#39; / &#39;danger&#39;</td></tr><tr><td>icon</td><td>string</td><td></td><td>否</td><td>图标，使用 <a href="./../components/Icon.html">Icon</a> 组件，图标优先级高于 <code>type</code>，颜色随 <code>type</code></td></tr><tr><td>duration</td><td>number</td><td>1500</td><td>否</td><td>展示时长(单位 ms)，值为 0 时，<code>notify</code> 不会消失</td></tr><tr><td>backgroundColor</td><td>string</td><td></td><td>否</td><td>指定背景颜色，会覆盖 type 的颜色</td></tr><tr><td>color</td><td>string</td><td></td><td>否</td><td>指定前景颜色（含字体，图标）</td></tr><tr><td>closable</td><td>boolean</td><td>false</td><td>是</td><td>是否显示关闭按钮</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>接口调用结束的回调函数（调用成功、失败都会执行）</td></tr></tbody></table><h4 id="object-type-的合法值" tabindex="-1">object.type 的合法值 <a class="header-anchor" href="#object-type-的合法值" aria-hidden="true">#</a></h4><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>primary</td><td>提示</td></tr><tr><td>success</td><td>成功</td></tr><tr><td>warning</td><td>警告</td></tr><tr><td>danger</td><td>危险</td></tr></tbody></table><h3 id="usage" tabindex="-1">Usage <a class="header-anchor" href="#usage" aria-hidden="true">#</a></h3><div class="language-"><pre><code>import { Notify } from &#39;vfox&#39;

Notify.showNotify({
  title: &#39;提示内容&#39;,
  duration: 2000
})
</code></pre></div><h2 id="notify-hidenotify-object" tabindex="-1">Notify.hideNotify([object]) <a class="header-anchor" href="#notify-hidenotify-object" aria-hidden="true">#</a></h2><p>隐藏消息提示框。</p><h3 id="params-1" tabindex="-1">Params <a class="header-anchor" href="#params-1" aria-hidden="true">#</a></h3><p>Object object</p><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败的回调函数</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>接口调用结束的回调函数（调用成功、失败都会执行）</td></tr></tbody></table><h3 id="usage-1" tabindex="-1">Usage <a class="header-anchor" href="#usage-1" aria-hidden="true">#</a></h3><div class="language-"><pre><code>import { Notify } from &#39;vfox&#39;

Notify.hideNotify()
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/apis/LocalStorage" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>LocalStorage</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/apis/Toast" data-v-38ede35f><span class="text" data-v-38ede35f>Toast</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_drawer.md\":\"e0e6bf7f\",\"apis_intersectionobserver.md\":\"ae842e28\",\"apis_localstorage.md\":\"641164a7\",\"apis_modal.md\":\"6554cfeb\",\"apis_notify.md\":\"36092315\",\"apis_popover.md\":\"a4ccb1b8\",\"apis_preview.md\":\"af8c6a1b\",\"apis_scroll.md\":\"702ef78a\",\"apis_selectorquery.md\":\"541f51aa\",\"apis_toast.md\":\"85ace50e\",\"apis_index.md\":\"07bf5529\",\"components_actionsheet.md\":\"4f543f5a\",\"components_activityindicator.md\":\"1f49bee6\",\"components_avatar.md\":\"4dd7655c\",\"components_backtop.md\":\"eccb7557\",\"components_badge.md\":\"bb0faad7\",\"components_button.md\":\"99d935d9\",\"components_calendar.md\":\"9ac29767\",\"components_cascader.md\":\"b775536d\",\"components_cell.md\":\"d9a64962\",\"components_checkbox.md\":\"194df522\",\"components_circleprogress.md\":\"12907ea5\",\"components_collapse.md\":\"3f7255cc\",\"components_copy.md\":\"2d4f685f\",\"components_countdown.md\":\"aeb73820\",\"components_countup.md\":\"f915b80e\",\"components_datepicker.md\":\"414dfb7f\",\"components_dialog.md\":\"aa441464\",\"components_divider.md\":\"bd8da4cf\",\"components_drawer.md\":\"461fdc5c\",\"components_dropdown.md\":\"fbbf4f71\",\"components_empty.md\":\"8a9f2cd0\",\"components_fixed.md\":\"99cd5ec2\",\"components_flatlist.md\":\"24f78132\",\"components_form.md\":\"2725e177\",\"components_group.md\":\"c8fbd653\",\"components_icon.md\":\"e0f24de3\",\"components_image.md\":\"1d2cacf7\",\"components_imagepreview.md\":\"df23288a\",\"components_imageuploader.md\":\"925929ca\",\"components_indexview.md\":\"ab42c661\",\"components_input.md\":\"9660a02f\",\"components_layout.md\":\"ab744be4\",\"components_loadmore.md\":\"1efe6534\",\"components_modal.md\":\"c380e91a\",\"components_navbar.md\":\"9a3cd985\",\"components_noticebar.md\":\"92c33f43\",\"components_notify.md\":\"a4d2b73f\",\"components_numberkeyboard.md\":\"bf56e8d4\",\"components_order.md\":\"512d25b1\",\"components_pagination.md\":\"41704043\",\"components_picker.md\":\"e76b8399\",\"components_popdialog.md\":\"cc503ee3\",\"components_popmenu.md\":\"5c07ed6c\",\"components_popover.md\":\"d04f3a46\",\"components_price.md\":\"cfe1f6ac\",\"components_progress.md\":\"544fa30b\",\"components_radio.md\":\"192504e4\",\"components_range.md\":\"94f518f6\",\"components_rate.md\":\"63dbf36b\",\"components_result.md\":\"987f262c\",\"components_scrolltab.md\":\"bf40d67f\",\"components_scrollview.md\":\"162e6a15\",\"components_searchbar.md\":\"328dd063\",\"components_sidetab.md\":\"8921cf78\",\"components_skeleton.md\":\"e0353755\",\"components_slider.md\":\"bc99c36d\",\"components_stepper.md\":\"d9517e78\",\"components_steps.md\":\"4407b183\",\"components_sticky.md\":\"bf12cbc9\",\"components_stopwatch.md\":\"4aa2a2ff\",\"components_swipecell.md\":\"b4b9dc77\",\"components_swiper.md\":\"83740298\",\"components_switch.md\":\"2a06b2af\",\"components_tab.md\":\"ff19cbc3\",\"components_tabbar.md\":\"383c5c64\",\"components_tabview.md\":\"c0ade8c1\",\"components_tag.md\":\"68d66898\",\"components_timeago.md\":\"176de266\",\"components_timeline.md\":\"0e941581\",\"components_toast.md\":\"d525633a\",\"components_index.md\":\"7a669029\",\"design_color.md\":\"998daa3b\",\"design_index.md\":\"a42fb833\",\"index.md\":\"65f5cb03\"}")</script>
    <script type="module" async src="/vfox/assets/app.a39fe1d2.js"></script>
    
  </body>
</html>