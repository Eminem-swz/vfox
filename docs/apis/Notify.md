# Notify 消息提示

注：目前仅支持单例模式。

## showNotify(object)

显示消息提示。

### object

| 属性            | 类型     | 默认值    | 必填 | 说明                                                                                |
| --------------- | -------- | --------- | ---- | ----------------------------------------------------------------------------------- |
| title           | string   |           | 是   | 提示内容                                                                            |
| type            | string   | 'default' | 否   | 提示类型，可选值：'primary' / 'success' / 'warning' / 'danger'                      |
| icon            | string   |           | 否   | 图标，使用 [Icon](../components/Icon.md) 组件，图标优先级高于 `type`，颜色随 `type` |
| duration        | number   | 1500      | 否   | 展示时长(单位 ms)，值为 0 时，`notify` 不会消失                                     |
| backgroundColor | string   |           | 否   | 指定背景颜色，会覆盖 type 的颜色                                                    |
| color           | string   |           | 否   | 指定前景颜色（含字体，图标）                                                        |
| closable        | boolean  | false     | 是   | 是否显示关闭按钮                                                                    |
| success         | function |           | 否   | 接口调用成功的回调函数                                                              |
| fail            | function |           | 否   | 接口调用失败的回调函数（不传入 fail 遇错误直接抛出）                                |
| complete        | function |           | 否   | 接口调用结束的回调函数（调用成功、失败都会执行）                                    |

#### object.type 的合法值

| 值      | 说明 |
| ------- | ---- |
| primary | 提示 |
| success | 成功 |
| warning | 警告 |
| danger  | 危险 |

### Usage

具体调用方式可以参考[API 调用](../guide/import.md#api-调用)。

```JavaScript
showNotify({
  title: '提示内容',
  duration: 2000
})
```

## hideNotify([object])

隐藏消息提示框。

### object

| 属性     | 类型     | 默认值 | 必填 | 说明                                             |
| -------- | -------- | ------ | ---- | ------------------------------------------------ |
| success  | function |        | 否   | 接口调用成功的回调函数                           |
| fail     | function |        | 否   | 接口调用失败的回调函数                           |
| complete | function |        | 否   | 接口调用结束的回调函数（调用成功、失败都会执行） |

### Usage

具体调用方式可以参考[API 调用](../guide/import.md#api-调用)。

```JavaScript
hideNotify()
```
