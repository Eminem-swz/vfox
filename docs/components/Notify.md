# Notify 消息提示

## Import

```
import { Notify } from 'vfox'
```

具体的引入方式可以参考[引入组件](../guide/import.md)。

## Props

| 属性            | 类型    | 默认值    | 必填 | 说明                                                                              |
| --------------- | ------- | --------- | ---- | --------------------------------------------------------------------------------- |
| v-model:visible | boolean | false     | 是   | 是否显示                                                                          |
| closable        | boolean | false     | 是   | 是否显示关闭按钮                                                                  |
| icon            | string  |           | 否   | 图标，使用 [Icon](./Icon.md) 组件                                                 |
| type            | string  | 'default' | 否   | 提示类型，可选值：'primary', 'success', 'warning', 'danger'                       |
| duration        | number  | 0         | 否   | visible=true 展示后，duration 毫秒后消失，0 为不消失，在 `v-model:visible` 下生效 |
| backgroundColor | string  |           | 否   | 指定背景颜色，会覆盖 type 的颜色                                                  |
| color           | string  |           | 否   | 指定前景颜色颜色（含字体，图标），会覆盖 type 的颜色                              |

### type 的合法值

| 值      | 说明     |
| ------- | -------- |
| primary | 提示     |
| success | 成功     |
| warning | 警告     |
| danger  | 强烈警告 |

## Events

| 事件                 | 描述               | 回调函数参数                                               |
| -------------------- | ------------------ | ---------------------------------------------------------- |
| close-click          | 关闭按钮点击时触发 | {}                                                         |
| visible-state-change | 展示隐藏时触发     | { state: [VisibleState](./Notify.md#visiblestate-值说明) } |

### VisibleState 值说明

| 值     | 说明                 | 备注                                              |
| ------ | -------------------- | ------------------------------------------------- |
| show   | 展示时触发           |                                                   |
| shown  | 展示且动画结束后触发 |                                                   |
| hide   | 隐藏时触发           | 可能携带其他参数 cancel, maskClick, closeClick 等 |
| hidden | 隐藏且动画结束后触发 | 可能携带其他参数 cancel, maskClick, closeClick 等 |
