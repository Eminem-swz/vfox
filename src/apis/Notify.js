import Vue from 'vue'
import Notify from '../components/Notify'
import { isObject, isString } from '../helpers/util'
import { parseParamsByRules } from './rules'
import { getCallbackFns } from './callback'
import Exception from '../helpers/exception'
import { removeEl } from '../helpers/dom'
import { createPopup } from '../helpers/popup'

let timers = {}
let $els = {}

export function baseShow(object, apiName, componentOptions) {
  if (isString(object)) {
    object = {
      title: object
    }
  } else if (!isObject(object)) {
    object = {}
  }

  const { success, fail, complete } = getCallbackFns(object)

  try {
    const key = apiName.replace('show', '')
    const propsData = parseParamsByRules(object, apiName)

    _hide(true, key)

    const Comp = Vue.extend({
      extends: componentOptions,
      methods: {
        close() {
          _hide(false, key)
        }
      },
      mounted() {
        setTimeout(() => {
          this.visible = true
        }, 17)
      }
    })

    const { $wrapper, zIndex } = createPopup()

    const app = new Comp({
      propsData: Object.assign(propsData, {
        zIndex
      })
    }).$mount($wrapper)

    $els[key] = app

    success({})
  } catch (e) {
    fail(e)
  }
  complete()
}

/**
 * 展示消息提示
 * @param {{title: String, type?: String, duration?: Number, backgroundColor?: String, color?: String}} object 配置
 */
export function showNotify(object) {
  return baseShow(object, 'showNotify', Notify)
}

function _hide(immediate = false, key) {
  clearTimeout(timers[key])

  if ($els[key]) {
    const app = $els[key]

    app.visible = false

    if (immediate) {
      app.$destroy()
      removeEl(app.$el)
    } else {
      setTimeout(() => {
        app.$destroy()
        removeEl(app.$el)
      }, 200)
    }

    delete $els[key]
  }
}

/**
 * 隐藏消息提示
 * @param {Object} object
 */
export function baseHide(object, apiName) {
  if (!isObject(object)) {
    object = {}
  }

  const { success, fail, complete } = getCallbackFns(object)

  try {
    _hide(false, apiName.replace('hide', ''))

    success()
  } catch (e) {
    fail(new Exception(e.message))
  }
  complete()
}

/**
 * 隐藏消息提示
 * @param {Object} object
 */
export function hideNotify(object) {
  return baseHide(object, 'hideNotify')
}
