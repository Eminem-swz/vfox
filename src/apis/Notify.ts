import { createApp } from 'vue'
import Notify from '@/Notify'
import { isObject, isString } from '@/helpers/util'
import { parseParamsByRules } from '@/apis/rules'
import { getCallbackFns } from '@/apis/callback'
import Exception from '@/helpers/exception'
import { createPopup, PopupBridge } from '@/utils/popup'
import { ApiOptions } from '@/apis/types'
import { StateTypes } from '@/utils/constants'

type ShowNotifyApiOptions = {
  title: string
  type?: StateTypes
  icon?: string
  duration?: number
  backgroundColor?: string
  color?: string
  closable?: boolean
} & ApiOptions

interface RefFns {
  [propName: string]: Function
}

const $refs: {
  [propName: string]: {
    uid: number
    fns: RefFns
  }
} = {}

export function baseShow(
  object: string | ApiOptions,
  apiName: string,
  componentOptions: any
) {
  if (isString(object)) {
    object = {
      title: object
    }
  } else if (!isObject(object)) {
    object = {}
  }

  const { success, fail, complete } = getCallbackFns(object as ApiOptions)

  try {
    const fns: RefFns = {}
    const key = apiName.replace('show', '')
    const propsData = parseParamsByRules(object as ApiOptions, apiName)

    clear(key)

    const { $wrapper } = createPopup()

    const app = createApp(
      componentOptions,
      Object.assign(propsData, {
        visible: true
      })
    )
    app.provide('fxApis', {
      in(hookName) {
        switch (hookName) {
          case 'afterHidden': {
            app.unmount($wrapper)
            remove(key, $ref.uid)
          }
        }
      },
      out(key: string, value: any) {
        fns[key] = value
      }
    } as PopupBridge)
    app.mount($wrapper)

    const $ref = {
      uid: app._uid,
      fns
    }

    $refs[key] = $ref

    success({})
  } catch (e) {
    fail(e)
  }
  complete()
}

function clear(key: string) {
  if ($refs[key]) {
    $refs[key].fns.customCancel('clear', true)
    delete $refs[key]
  }
}

function remove(key: string, uid: number) {
  if ($refs[key] && $refs[key].uid === uid) {
    delete $refs[key]
  }
}

/**
 * 展示消息提示
 * @param object 配置
 */
export function showNotify(object: ShowNotifyApiOptions) {
  return baseShow(object, 'showNotify', Notify)
}

/**
 * 隐藏消息提示
 * @param {Object} object
 */
export function baseHide(object: ApiOptions, apiName: string) {
  if (!isObject(object)) {
    object = {}
  }

  const { success, fail, complete } = getCallbackFns(object)

  try {
    clear(apiName.replace('hide', ''))

    success()
  } catch (e) {
    fail(new Exception(e.message))
  }
  complete()
}

/**
 * 隐藏消息提示
 * @param {Object} object
 */
export function hideNotify(object: ApiOptions) {
  return baseHide(object, 'hideNotify')
}
