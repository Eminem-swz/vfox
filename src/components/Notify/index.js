import Notify from './Notify.vue'

Notify.install = function(Vue) {
  Vue.component(Notify.name, Notify)
}

export default Notify
