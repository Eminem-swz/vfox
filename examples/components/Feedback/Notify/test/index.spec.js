/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Notify from '@/Notify'

describe('Notify', () => {
  test('should render default correctly', () => {
    const wrapper = mount(Notify, {
      propsData: {
        title: '通知文本'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop type="success" correctly', () => {
    const wrapper = mount(Notify, {
      propsData: { title: '成功文本', type: 'success' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop type="warning" correctly', () => {
    const wrapper = mount(Notify, {
      propsData: { title: '警告文本', type: 'warning' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop type="danger" correctly', () => {
    const wrapper = mount(Notify, {
      propsData: { title: '危险文本', type: 'danger' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
