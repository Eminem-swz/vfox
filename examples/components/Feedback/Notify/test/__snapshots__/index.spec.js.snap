// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notify should render default correctly 1`] = `
<div class="fx-notify fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-notice-bar type--primary fx-notify_inner">
    <!---->
    <div class="fx-notice-bar_content">
      <div class="fx-notice-bar_content-inner">
        通知文本
      </div>
    </div>
    <!---->
  </div>
</div>
`;

exports[`Notify should render prop type="danger" correctly 1`] = `
<div class="fx-notify fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-notice-bar type--danger fx-notify_inner">
    <!---->
    <div class="fx-notice-bar_content">
      <div class="fx-notice-bar_content-inner">
        危险文本
      </div>
    </div>
    <!---->
  </div>
</div>
`;

exports[`Notify should render prop type="success" correctly 1`] = `
<div class="fx-notify fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-notice-bar type--success fx-notify_inner">
    <!---->
    <div class="fx-notice-bar_content">
      <div class="fx-notice-bar_content-inner">
        成功文本
      </div>
    </div>
    <!---->
  </div>
</div>
`;

exports[`Notify should render prop type="warning" correctly 1`] = `
<div class="fx-notify fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-notice-bar type--warning fx-notify_inner">
    <!---->
    <div class="fx-notice-bar_content">
      <div class="fx-notice-bar_content-inner">
        警告文本
      </div>
    </div>
    <!---->
  </div>
</div>
`;
